src/router/BodLayerId.routerPlugin.js

```diff
@@ -0,0 +1,236 @@
+ *   the showtooltip parameter if it was set
+ */
+function parseBodLayerParamsFromQuery(query, layersConfig) {
+    if (!query) {
+        return undefined
+    }
+    const nonStandardParamsPresence = Array.from(Object.keys(query)).some(
+        (param) => !standardsParams.includes(param)
+    )
+    // if all parameters are standard, we do not need to be here
+    if (!nonStandardParamsPresence) {
+        return undefined
+    }
+    const val = {}
+    Object.entries(query).forEach(([key, value]) => {
+        let layer = null
+        if (!standardsParams.includes(key)) {
+            layer = layersConfig.find((layer) => layer.getID() === key)
+        }
+        if (layer) {
+            const clonedLayer = layer.clone()
+            clonedLayer.visibility = true
+            val.layer = clonedLayer
+            val.features = value.split(',')
+            // for the showToolTip parameter : it should be false only when specified as false
+            val.showToolTip = !(query.showtooltip === 'false' || query.showtooltip === false)
+            return null // we get out at the first instance of a bod layer id found
+        }
+    })
+    // if we found no layer, we return undefined
+    return val?.layer ? val : undefined
+}
+/**
+ * Checks the layers param in the URL and either ensure an already activated layer is visible, or
+ * add the layer in `bodIdParams.layer` to the layers param. It also dispatch the highlighted
+ * features to the store.
+ *
+ * If there is no layers params, this means the only active layer will be the bod Id layer given
+ *
+ * @param {Object} bodIdParams All the bod-layer-id parameters needed
+ * @param {Store} store
+ * @param {any} to
+ * @param {any} next
+ */
+function handleBodIdParams(bodIdParams, store, to, next) {
+    const newQuery = { ...to.query }
+    // QUESTION TO pascal : do we allow the bod layer id param to have the same parameters as the other layers ?
+    // in my opinion : no, because it's simpler, but it might be done
+
+    if (newQuery.layers?.includes(bodIdParams.layer.getID())) {
+        const regex = new RegExp(bodIdParams.layer.getID() + ',[a-zA-Z]+')
+        // this will replace the `{layer-id},{visibility}` parameter by `{layer-id},`
+        // forcing the layer to be visible
+        newQuery.layers = newQuery.layers.replace(regex, `${bodIdParams.layer.getID()},`)
+    } else {
+        // the parameter given was not part of the layers, or there was no active layers
+        newQuery.layers = `${
+            newQuery.layers ? newQuery.layers + ';' : ''
+        }${bodIdParams.layer.getID()}`
+    }
+    delete newQuery[bodIdParams.layer.getID()]
+    delete newQuery.showtooltip
+    // the center parameter is always overwritten
+    delete newQuery.center
+
+    retrieveAndDispatchFeaturesToStore(bodIdParams, !newQuery.z, store)
+
+    next({
+        name: 'MapView',
+        query: newQuery,
+    })
```

Comment:
I would not try add the layer in the `layers` url parameter here, that's the job of the `LayersParamParser`. Simply check if the layer is in the active layers in store and if not dispatch a `addLayer` with the layer ID. We don't need to force the visible flag as it is not forced on the old viewer when the layer is in active layer list but invisible.